src/main/java/de/craften/plugins/rpgplus/components/entitymanager/RpgPlusEntity.java

```diff
@@ -2,39 +2,143 @@
 
 import net.citizensnpcs.api.CitizensAPI;
 import net.citizensnpcs.api.ai.TargetType;
+import net.citizensnpcs.api.event.NPCDeathEvent;
 import net.citizensnpcs.api.npc.NPC;
+import net.citizensnpcs.api.trait.Trait;
 
 import org.bukkit.Location;
+import org.bukkit.entity.ArmorStand;
 import org.bukkit.entity.Creature;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.EntityType;
 import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.EventHandler;
 import org.bukkit.event.player.PlayerTeleportEvent;
 
+import de.craften.plugins.rpgplus.util.EntityUtil;
+
 /**
  * A basic managed entity without any special logic.
  */
 public class RpgPlusEntity<T extends Entity> {
     private final NPC npc;
     private Location location;
     private boolean isTakingDamage = true;
-
+    
+    private EntityType type;
+    
+    private boolean isNameVisible;
+    private String name;
+    private String secondName;
+    private ArmorStand nametag;
+    private ArmorStand secondNameTag;
+    
     public RpgPlusEntity(Location location, EntityType type) {
         this.location = location;
+        this.type = type;
         npc = CitizensAPI.getNPCRegistry().createNPC(type, "");
-        npc.data().set(NPC.NAMEPLATE_VISIBLE_METADATA, true);
+        
+        if (type == EntityType.PLAYER) {
+        	npc.data().set(NPC.NAMEPLATE_VISIBLE_METADATA, true);
+        } else {
+        	npc.data().set(NPC.NAMEPLATE_VISIBLE_METADATA, false);
+        }
+        
     }
 
     public T spawn() {
         // TODO support custom skins
         // npc.data().setPersistent(NPC.PLAYER_SKIN_UUID_METADATA, "leMaik");
 
+        npc.addTrait(new Trait("NameTagTrait") {
```

Comment:
This trait should be extracted into a separate file, like the other traits (e.g. `HorseTrait`).